config/rc.lua

```diff
@@ -28,6 +28,9 @@ if awesome.startup_errors then
                      title = "Oops, there were errors during startup!",
                      text = awesome.startup_errors })
 end
+naughty.notify({ preset = naughty.config.presets.critical,
```

Comment:
Did you just duplicate this by accident??

Reply:
I put this in here to test notifications. I was _hoping_ to get them working when I first started on this PR (notice a lot of the changes relate to adding cairo and xcb stuff to awesome bindings). Alas, no such luck. I'll remove this as this is a debug thing I forgot to remove.

I'm going to review this again properly and merge it in soon (tm). There's some changes here that are fairly critical (fix a lot of issues with Awesome classes that will be needed down the line. Plus it shows how to connect and use xcb)

Reply:
That would be great! I was going to base the wallpaper work off of this branch anyway!

Reply:
I'll work on cleaning this up and merging it in today.

Reply:
Ok "today" was a bit optimistic. Going to try to finish this up ASAP, but you should still base your work off this branch.